src/transformers/models/omdet_turbo/modeling_omdet_turbo.py

```diff
@@ -0,0 +1,1916 @@
+        self.linear2 = nn.Linear(d_ffn, encoder_hidden_dim)
+        self.dropout4 = nn.Dropout(dropout)
+        self.norm3 = nn.LayerNorm(encoder_hidden_dim)
+
+        self.fuse_type = fuse_type
+
+    @staticmethod
+    def with_pos_embed(tensor, pos):
+        return tensor if pos is None else tensor + pos
+
+    def forward_ffn(self, tgt):
+        tgt2 = self.linear2(self.dropout3(self.act(self.linear1(tgt))))
+        tgt = tgt + self.dropout4(tgt2)
+        tgt = self.norm3(tgt)
+        return tgt
+
+    def forward(self, embed, task_feats, refer_bbox, feats, shapes, padding_mask=None, attn_mask=None, query_pos=None):
+        origin_emb_len = embed.shape[1]
+
+        # self attention
+        q = k = self.with_pos_embed(embed, query_pos)
+
+        # combine task_emb with q, k, v
+        if self.fuse_type == "merged_attn":
+            task_feats = task_feats.transpose(0, 1)  # [bs, token_len, hidden]
+            q = torch.cat((q, task_feats), dim=1)  # [bs, dn+num_query+token_len, hidden]
+            k = torch.cat((k, task_feats), dim=1)  # [bs, dn+num_query+token_len, hidden]
+            embed = torch.cat((embed, task_feats), dim=1)  # [bs, dn+num_query+token_len, hidden]
+
+        tgt = self.self_attn(q.transpose(0, 1), k.transpose(0, 1), embed.transpose(0, 1), attn_mask=attn_mask)[
+            0
+        ].transpose(0, 1)
+        embed = embed + self.dropout1(tgt)
+        embed = self.norm1(embed)
+
+        # cut fused embedd to vision emb and task emb         todo  spilt here or split before
+        task_feats = embed[:, origin_emb_len:, :].transpose(0, 1)  # [token_len, bs, hidden]
+        embed = embed[:, :origin_emb_len, :]  # [bs, dn+num_query, hidden]
+
+        # cross attention
+        tgt = self.cross_attn(
+            hidden_states=self.with_pos_embed(embed, query_pos),
+            attention_mask=padding_mask,
+            encoder_hidden_states=feats,
+            reference_points=refer_bbox.unsqueeze(2),
+            spatial_shapes=torch.tensor(shapes, dtype=torch.int64),
+            # self.with_pos_embed(embed, query_pos), refer_bbox.unsqueeze(2), feats, shapes, padding_mask
+        )[0]
+        embed = embed + self.dropout2(tgt)
+        embed = self.norm2(embed)
+
+        # ffn
+        embed = self.forward_ffn(embed)
+
+        return embed, task_feats
+
+
+class OmDetTurboDeformableTransformerDecoder(nn.Module):
+    """
+    https://github.com/PaddlePaddle/PaddleDetection/blob/develop/ppdet/modeling/transformers/deformable_transformer.py
+    """
+
+    def __init__(self, hidden_dim, decoder_layer, num_layers, eval_idx=-1, cls_type="cosine"):
+        super().__init__()
+        self.layers = self._get_clones(decoder_layer, num_layers)
+        self.num_layers = num_layers
+        self.hidden_dim = hidden_dim
+        self.eval_idx = eval_idx if eval_idx >= 0 else num_layers + eval_idx
+        self.cls_type = cls_type
+        self.logit_scale = torch.ones([]) * np.log(1 / 0.07)
```

Comment:
Is it a constant or should depend on logit shape?

Reply:
Looks like a constant from the original implementation